sql/moz-fx-data-shared-prod/telemetry_dev_cycle_derived/glean_metrics_external_v1/query.py

```diff
@@ -0,0 +1,122 @@
+"""Glean metric data - download from API, clean and upload to BigQuery."""
+
+import logging
+
+import click
+import requests
+from google.api_core.exceptions import BadRequest
+from google.cloud import bigquery
+
+API_BASE_URL = "https://probeinfo.telemetry.mozilla.org"
+SCHEMA = [
+    bigquery.SchemaField("glean_app", "STRING"),
+    bigquery.SchemaField("metric", "STRING"),
+    bigquery.SchemaField("type", "STRING"),
+    bigquery.SchemaField("first_seen_date", "DATE"),
+    bigquery.SchemaField("last_seen_date", "DATE"),
+    bigquery.SchemaField("expires", "STRING"),
+]
+PROJECT_ID = "moz-fx-data-shared-prod"
+DATASET_ID = "telemetry_dev_cycle_derived"
+TABLE_NAME = "glean_metrics_external_v1"
+
+
+def get_api_response(url):
+    """Get json of response if the requests to the API was successful."""
+    response = requests.get(url)
+
+    if response.status_code != 200:
+        logging.error(
+            f"Failed to download data from {url}. \nResponse status code {response.status_code}."
+        )
+        return
```

Comment:
We should probably error out at this point. Just returning None here would suggest to me that this is expected outcome and that it is fine. But looking at the caller it seems to assume a valid response that it can process it. Right now if we return None and then try to iterate through it we will just return an empty list and program will continue execution and complete without a failure. This could potentially result in silent failures where the task is showing as success even though it is actually failing.

Reply:
https://github.com/mozilla/bigquery-etl/pull/4720/files#diff-0e5f2ccbecfd0a9a76eae401752509326f63daac95b39a781708b77899fb3222R54

In here though, we'd just not enter the loop and exit without doing anything?

Reply:
yeah i changed it ... it's now throwing an error so should be fine and catches the error if i want to proceed anyway